rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -542,9 +542,11 @@ def build(self, *, doc_build_folder, output_staging_directory):
         ]
 
         build_context = self.build_context
-        has_python = build_context.build_type == 'ament_python' or \
-            build_context.always_run_sphinx_apidoc or \
-            build_context.ament_cmake_python
+        has_python = (
+            build_context.build_type == 'ament_python'
+            or build_context.always_run_sphinx_apidoc
+            or build_context.ament_cmake_python) \
+            and not build_context.never_run_sphinx_apidoc
```

Comment:
Can we separate this out so that this condition triggers a helpful debug message like the doxygen one above?

Reply:
Done.